samples/CSharp/SourceGenerators/SourceGeneratorSamples/MathsGenerator.cs

```diff
@@ -0,0 +1,474 @@
+
+                Term(ctx);
+            }
+        }
+        private static void Term(Context ctx) {
+            // term    = factor { (times|divide) factor };
+            Factor(ctx);
+
+            while(IsOp(ctx, "*") || IsOp(ctx, "/")) {
+                if(IsOp(ctx, "*")) ConsOp(ctx, "*");
+                if(IsOp(ctx, "/")) ConsOp(ctx, "/");
+
+                Term(ctx);
+            }
+        }
+        private static void Factor(Context ctx) {
+            // factor  = number | var | func | lround expr rround;
+            if(Peek(ctx, TokenType.Number)) {
+                Consume(ctx, TokenType.Number);
+                return;
+            }
+
+            if(Peek(ctx, TokenType.Identifier)) {
+                Consume(ctx, TokenType.Identifier); // Is either var or func
+                if(Peek(ctx, TokenType.OpenParens, "(")) { // Is Func, but we already consumed its name
+                    Funct(ctx);
+                }
+                return;
+            }
+            if(Peek(ctx, TokenType.Sum)) {
+                Sum(ctx);
+                return;
+            }
+            // Must be a parenthesized expression
+            Consume(ctx, TokenType.OpenParens);
+            Expr(ctx);
+            Consume(ctx, TokenType.CloseParens);
+        }
+        private static void Sum(Context ctx) {
+            // sum     = ∑ lround identifier comma expr1 comma expr2 comma expr3 rround;
+            // TODO: differentiate in the language between integer and double, but complicated for a sample.
+            Consume(ctx, TokenType.Sum);
+            Consume(ctx, TokenType.OpenParens, "(");
+
+            AddSymbol(ctx);
+            var varName = NextToken(ctx).Value;
+            NextToken(ctx); // consume the first comma without emitting it
+
+            ctx.buffer.Append("(int)");
+            Expr(ctx); // Start index
+            Consume(ctx, TokenType.Comma);
+
+            ctx.buffer.Append("(int)");
+            Expr(ctx); // End index
+            Consume(ctx, TokenType.Comma);
+
+            ctx.buffer.Append($"{varName} => "); // It needs to be a lambda
+            
+            Expr(ctx); // expr to evaluate at each iteration
+
+            Consume(ctx, TokenType.CloseParens, ")");
+        }
+        private static void Funct(Context ctx) {
+            // func    = identifier lround expr {comma expr} rround;
+            Consume(ctx, TokenType.OpenParens, "(");
+            Expr(ctx);
+            while(Peek(ctx, TokenType.Comma)) {
+                Consume(ctx, TokenType.Comma);
+                Expr(ctx);
+            }
+            Consume(ctx, TokenType.CloseParens, ")");
+        }
+    }
+
+    [Generator]
+    public class MathsGenerator : ISourceGenerator
+    {
+        private bool libraryIsAdded = false;
+
+        private const string libraryCode = @"
+using System.Linq;
+using System;
+using System.Collections.Generic;
+
+namespace ___MathLibrary___ {
```

Comment:
Hmm, I think I get what you're trying to do here (avoid conflicting with user defined types) and yeah we don't have hygenic macros / names in the language. Same time the `__` in the name throws me off a bit. 

Is there a reason you just didn't make it a util style class, like `FormulaHelpers`, inside the `Math` namepace that you're generating?

Reply:
Fixed in 425941d959a4c69588f79e7407841ab1e97fba0a. Great point about hygienic macros. I didn't realize the similarity. I need to remember to touch on that in the blog post.